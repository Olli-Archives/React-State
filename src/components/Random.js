import React, { PureComponent } from 'react';
import ColorDiv from './ColorDiv';
import Chance from 'chance';
const chance = new(Chance);

export default class RandomColor extends PureComponent{
state = {
  color:''
}

newColor = ()=>{
  const color = chance.color({ format: 'hex' });
  console.log('hi', this.state.color);
  this.setState({
    color:color
  });
}



componentDidMount(){
  this.myInterval = setInterval(this.newColor, 2000); 
}
componentWillUnmount(){
  clearInterval(this.myInterval);
}
render(){
  return (
    <div>
      <ColorDiv color={this.state.color}/>
    </div>
  );
}

}
