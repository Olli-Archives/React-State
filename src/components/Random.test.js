import React from 'react';
import { shallow } from 'enzyme';
import Random from './Random';

describe('random component', ()=>{
  it('renders Random', ()=>{
    const wrapper = shallow(<Random/>);
    expect(wrapper).toMatchSnapshot();
  });
})
;
