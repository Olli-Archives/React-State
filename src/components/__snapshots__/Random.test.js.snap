// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`random component renders Random 1`] = `
<div>
  <ColorDiv
    color=""
  />
</div>
`;
